pkg/agent/agent.go

```diff
@@ -369,16 +381,36 @@ func updateAgentStatus(status, message string) error {
 	if agent.agentResource != nil {
 		agentResourceType := getAgentResourceType()
 		resource := createAgentStatusSubResource(agentResourceType, status, message)
-		buffer, err := json.Marshal(resource)
-		if err != nil {
-			return nil
+
+		// Check if there is a generic resource to update
+		var genResource interface{}
+		if agentResourceType == v1alpha1.AWSDiscoveryAgentResource || agentResourceType == v1alpha1.EdgeDiscoveryAgentResource {
+			genResource = createAgentStatusSubResource(v1alpha1.DiscoveryAgentResource, status, message)
+		} else if agentResourceType == v1alpha1.AWSTraceabilityAgentResource || agentResourceType == v1alpha1.EdgeTraceabilityAgentResource {
+			genResource = createAgentStatusSubResource(v1alpha1.TraceabilityAgentResource, status, message)
 		}
 
-		subResURL := agent.cfg.GetEnvironmentURL() + "/" + agentResourceType + "/" + agent.cfg.GetAgentName() + "/status"
-		_, err = agent.apicClient.ExecuteAPI(coreapi.PUT, subResURL, nil, buffer)
-		if err != nil {
-			return err
+		if genResource != nil {
+			err := updateAgentStatusAPI(resource, agentResourceType)
+			if err != nil {
+				log.Warn("Could not update the generic agent reference")
+			}
 		}
+		return updateAgentStatusAPI(resource, agentResourceType)
```

Comment:
if genResource != nil we update the status twice? Is 394 meant to be genResource?